static/app/components/searchQueryBuilder/tokens/filter/utils.tsx

```diff
@@ -68,3 +68,28 @@ export function formatFilterValue(token: TokenResult<Token.FILTER>['value']): st
       return token.text;
   }
 }
+
+/**
+ * Replaces the focused parametr (at cursorPosition) with the new value.
+ *
+ * Example:
+ * replaceCommaSeparatedValue('foo,bar,baz', 5, 'new') => 'foo,new,baz'
+ */
+export function replaceCommaSeparatedValue(
+  value: string,
+  cursorPosition: number | null,
+  replacement: string
+) {
+  const items = value.split(',');
+
+  let characterCount = 0;
+  for (let i = 0; i < items.length; i++) {
+    characterCount += items[i].length + 1;
+    if (characterCount > (cursorPosition ?? value.length + 1)) {
+      const newItems = [...items.slice(0, i), replacement, ...items.slice(i + 1)];
+      return newItems.map(item => item.trim()).join(',');
```

Comment:
I'm not sure where else this may get used, but trimming the item seems like a side effect compared to the function name and doc string. If it's possible to remove it that might be a good call but mentioning it works as well.

Reply:
I do agree with this actually, but I'll leave it for now since this is used for the string filter multi-select as well, which may need that trimming behavior (I forget if it was actually necessary or not). I may come back to this and remove that trim later on